src/OrchardCore/OrchardCore.Data.YesSql/DbConnectionValidator.cs

```diff
@@ -125,63 +156,29 @@ public async Task<DbConnectionValidatorResult> ValidateAsync(string databaseProv
 
         // The 'Document' table exists.
         return DbConnectionValidatorResult.DocumentTableFound;
-    }
-
-    private ISqlBuilder GetSelectBuilderForDocumentTable(string tablePrefix, DatabaseProviderName providerName)
-    {
-        var selectBuilder = GetSqlBuilder(providerName, tablePrefix);
 
-        selectBuilder.Select();
-        selectBuilder.Selector("*");
-        selectBuilder.Table(_tableNameConvention.GetDocumentTable());
-        selectBuilder.Take("1");
-
-        return selectBuilder;
-    }
-
-    private ISqlBuilder GetSelectBuilderForShellDescriptorDocument(string tablePrefix, DatabaseProviderName providerName)
-    {
-        var selectBuilder = GetSqlBuilder(providerName, tablePrefix);
-
-        selectBuilder.Select();
-        selectBuilder.Selector("*");
-        selectBuilder.Table(_tableNameConvention.GetDocumentTable());
-        selectBuilder.WhereAnd($"Type = '{_shellDescriptorTypeColumnValue}'");
-        selectBuilder.Take("1");
+        static ISqlBuilder GetDocumentTableSelectBuilder(string tablePrefix, string tablePrefixSeparator, string documentTable, ISqlDialect dialect, bool isShellDescriptorDocument = false)
```

Comment:
Can you move this outside of the `ValidateAsync` method and use the `private` keyword?

Reply:
Seems `GetDocumentTableSelectBuilder()` make it a little bit bigger, I will extract it